components/core/src/clp/ffi/ir_stream/search/test/test_deserializer_integration.cpp

```diff
@@ -0,0 +1,316 @@
+};
+
+/**
+ * Serializes a vector of JSON object pairs into a key-value pair IR stream.
+ * @tparam encoded_variable_t
+ * @param json_pairs A vector of JSON pairs. The objects in the pair are serialized as
+ * auto-generated kv-pairs and user-generated kv-pairs, respectively.
+ * @return A vector of bytes representing the serialized IR stream.
+ */
+template <typename encoded_variable_t>
+[[nodiscard]] auto serialize_json_pairs_into_kv_pair_ir_stream(
+        std::vector<JsonPair> const& json_pairs
+) -> std::vector<int8_t>;
+
+/**
+ * Serializes a vector of JSON object pairs into a string.
+ * @param json_pairs
+ * @return A string representing the serialized JSON pairs.
+ */
+[[nodiscard]] auto serialize_json_pairs_to_str(std::vector<JsonPair> const& json_pairs)
+        -> std::string;
+
+template <typename encoded_variable_t>
+auto serialize_json_pairs_into_kv_pair_ir_stream(std::vector<JsonPair> const& json_pairs)
+        -> std::vector<int8_t> {
+    auto serializer_result{Serializer<encoded_variable_t>::create()};
+    REQUIRE_FALSE(serializer_result.has_error());
+    auto& serializer{serializer_result.value()};
+    for (auto const& [auto_gen_kv_pairs, user_gen_kv_pairs] : json_pairs) {
+        REQUIRE(unpack_and_serialize_msgpack_bytes(
+                nlohmann::json::to_msgpack(auto_gen_kv_pairs),
+                nlohmann::json::to_msgpack(user_gen_kv_pairs),
+                serializer
+        ));
+    }
+    auto const ir_buf_view{serializer.get_ir_buf_view()};
+    std::vector<int8_t> ir_buf{ir_buf_view.begin(), ir_buf_view.end()};
+    ir_buf.emplace_back(static_cast<int8_t>(cProtocol::Eof));
+    return ir_buf;
+}
+
+auto serialize_json_pairs_to_str(std::vector<JsonPair> const& json_pairs) -> std::string {
+    std::string serialized_json_pairs;
+    for (auto const& [auto_gen, user_gen] : json_pairs) {
+        serialized_json_pairs += fmt::format(
+                "auto-gen-kv-pairs:{} | user-gen-kv-pairs:{}\n",
+                auto_gen.dump(),
+                user_gen.dump()
+        );
+    }
+    return serialized_json_pairs;
+}
+}  // namespace
+
+TEMPLATE_TEST_CASE(
+        "deserialization_kv_ir_stream_with_query",
+        "[ffi][ir_stream][search][QueryHandler]",
+        ir::four_byte_encoded_variable_t,
+        ir::eight_byte_encoded_variable_t
+) {
+    constexpr std::string_view cIntKey{"int_key"};
+    constexpr std::string_view cStrKey{"str_key"};
+    constexpr std::string_view cObjKey{"obj_key"};
+    constexpr std::string_view cUnresolvableKey{"unresolvable_key"};
+    constexpr int cIntBase{INT16_MAX};
+    constexpr std::string_view cTestSubStr{"test"};
+
+    nlohmann::json const base_obj_0
+            = {{cIntKey, cIntBase - 1}, {cStrKey, fmt::format("VarStr:{}", cTestSubStr)}};
+    nlohmann::json const base_obj_1 = {
+            {cIntKey, cIntBase + 1},
+            {cStrKey,
+             fmt::format("This is a ClpStr: {}; And it contains a var={}", cTestSubStr, cIntBase)}
```

Comment:
```suggestion
             fmt::format("This is a {} ClpStr; And it contains a var={}", cTestSubStr, cIntBase)}
```

Otherwise it looks like it's saying "test" is the ClpSttr.